src/Infrastructure/IO/src/ESMCI_PIO_Handler.C

```diff
@@ -603,37 +603,30 @@ void PIO_Handler::arrayRead(
     if (ESMC_LogDefault.MsgFoundError (ESMF_RC_FILE_READ, "file not open",
         ESMC_CONTEXT, rc)) return;
 
-  iodesc = getIODesc(pioSystemDesc, arr_p, &ioDims, &nioDims,
-      &arrDims, &narrDims, &basepiotype, &localrc);
-  if (ESMC_LogDefault.MsgFoundError(localrc, ESMCI_ERR_PASSTHRU,
-      ESMC_CONTEXT, rc)) return;
-
   // Get a pointer to the array data
   // Still have the one DE restriction so use localDE = 0
   localDE = 0;
   baseAddress = arr_p->getLocalarrayList()[localDE]->getBaseAddr();
   int arrlen = 1;
-  const int *counts = arr_p->getLocalarrayList()[localDE]->getCounts();
-  for (int i=0; i<narrDims; i++)
-      arrlen *= counts[i]; 
-
 #if defined(ESMF_NETCDF) || defined(ESMF_PNETCDF)
     int nDims;
-    if (((char *)NULL != name) && (strlen(name) > 0)) {
```

Comment:
@jedwards4b why was this part of the code removed?  I pushed a change just below this in the file to correctly set up varname.  Without it the varname is not being passed to PIOc_inq_varid below.  What's the right solution here?

Reply:
I think that it may have been accidental.  I don't recall removing it.